R/dplyr.R

```diff
@@ -122,3 +122,123 @@ rename.zoomed_dm <- function(.data, ...) {
   replace_zoomed_tbl(.data, renamed_tbl, new_tracked_keys_zoom)
 }
 
+#' @export
+left_join.dm <- function(x, ...) {
+  abort_no_table_zoomed_dplyr("left_join")
+}
+
+#' @export
+left_join.zoomed_dm <- function(x, y, by = NULL, copy = FALSE, suffix = c(".x", ".y"), select = NULL, ...) {
+  if (nrow(cdm_get_filter(x) %>% filter(table == !!orig_name_zoomed(x)))) abort_no_filters_rename_select()
+  y_name <- as_string(enexpr(y))
+  join_data <- prepare_join(x, y_name, by, enexpr(select), suffix[1])
```

Comment:
Ideally we'd get rid of the line above and do

```suggestion
  join_data <- prepare_join(x, {{ y }}, by, {{ select }}, suffix[1])
```

with suitable `ensym()` and `enquo()` calls in `prepare_join()` .